webhooks/virtualmachineclass/v1alpha2/mutation/virtualmachineclass_mutator_unit_test.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2019-2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package mutation_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	vmopv1 "github.com/vmware-tanzu/vm-operator/api/v1alpha2"
+	"github.com/vmware-tanzu/vm-operator/test/builder"
+)
+
+func uniTests() {
+	Describe("Invoking Mutate", unitTestsMutating)
+}
+
+type unitMutationWebhookContext struct {
+	builder.UnitTestContextForMutatingWebhook
+	vmClass *vmopv1.VirtualMachineClass
+}
+
+func newUnitTestContextForMutatingWebhook() *unitMutationWebhookContext {
+	vmClass := builder.DummyVirtualMachineClassA2()
+	obj, err := builder.ToUnstructured(vmClass)
+	Expect(err).ToNot(HaveOccurred())
+
+	return &unitMutationWebhookContext{
+		UnitTestContextForMutatingWebhook: *suite.NewUnitTestContextForMutatingWebhook(obj),
+		vmClass:                           vmClass,
+	}
+}
+
+func unitTestsMutating() {
+	var (
+		ctx *unitMutationWebhookContext
+	)
+
+	BeforeEach(func() {
+		ctx = newUnitTestContextForMutatingWebhook()
+	})
+	AfterEach(func() {
+		ctx = nil
+	})
+
+	Describe("VirtualMachineClassMutator should admit updates when object is under deletion", func() {
+		Context("when update request comes in while deletion in progress ", func() {
+			It("should admit update operation", func() {
+				t := metav1.Now()
+				ctx.WebhookRequestContext.Obj.SetDeletionTimestamp(&t)
+				response := ctx.Mutate(&ctx.WebhookRequestContext)
+				Expect(response.Allowed).To(BeTrue())
+			})
+		})
+	})
+}
```

Comment:
I noticed the ControllerName related tests have been dropped. Intentional?.

Reply:
Just missed. Thanks.